### PR TITLE
CLOUDP-200329: Log podman not found and continue with deployment command

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-200329](https://jira.mongodb.org/browse/CLOUDP-200329)

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/podman/client.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ import (
 )
 
 var (
-	ErrPodmanNotFound  = errors.New("podman not found in your system, check requirements at http://docpage")
+	ErrPodmanNotFound  = errors.New("podman not found in your system, check requirements at https://dochub.mongodb.org/core/atlas-cli-deploy-local-reqs")
```
Comment (blva) [inline internal/podman/client.go:32]: [CLOUDP-200468](https://jira.mongodb.org/browse/CLOUDP-200468)

File: internal/cli/atlas/deployments/list.go [hunk 2]
```diff
@@ -196,12 +190,16 @@ func ListBuilder() *cobra.Command {
...
-			opts.podmanClient = podman.NewClient(log.IsDebugLevel(), log.Writer())
-			return opts.podmanClient.Ready(cmd.Context())
+			opts.PodmanClient = podman.NewClient(log.IsDebugLevel(), log.Writer())
+			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return opts.Run(cmd.Context())
 		},
+		PostRunE: func(cmd *cobra.Command, args []string) error {
+			opts.PostRunMessages()
```
Comment (fmenezes) [inline internal/cli/atlas/deployments/list.go:200]: nit: we have this pattern for post run, commands with --watch normally have it
```suggestion
			return opts.PostRun(cmd.Context())
```

Comment (blva) [reply internal/cli/atlas/deployments/list.go:200]: thank you! will add

File: internal/podman/client.go (hunk 1 above)
Comment (wtrocki) [reply internal/podman/client.go:32]: [pro tip] Common pattern for clients that would not support automatic updates is to use url shortener with the ability to swap links at any point.

Comment (wtrocki) [reply internal/podman/client.go:32]: No action needed.

Comment (gssbzn) [reply internal/podman/client.go:32]: @wtrocki FYI dochub is our version of that

File: internal/cli/atlas/deployments/list.go [hunk 3]
```diff
@@ -155,7 +154,7 @@ func (opts *ListOpts) Run(ctx context.Context) error {
 	}
 
 	mdbContainers, err := opts.getLocalDeployments(ctx)
-	if err != nil {
+	if err != nil && err != podman.ErrPodmanNotFound {
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/list.go:157]:
```suggestion
	if err != nil && !errros.Is(err, podman.ErrPodmanNotFound) {
```

File: internal/cli/atlas/deployments/list.go [hunk 4]
```diff
@@ -185,23 +183,27 @@ func ListBuilder() *cobra.Command {
...
+				opts.InitOutput(log.Writer(), listTemplate)); err != nil {
 				return err
 			}
 
 			opts.defaultSetter.OutWriter = cmd.OutOrStdout()
 
-			opts.podmanClient = podman.NewClient(log.IsDebugLevel(), log.Writer())
-			return opts.podmanClient.Ready(cmd.Context())
+			opts.PodmanClient = podman.NewClient(log.IsDebugLevel(), log.Writer())
+			return nil
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/list.go:199]: why did we remove `opts.podmanClient.Ready(cmd.Context())`

File: internal/cli/atlas/deployments/options/deployment_opts.go [hunk 5]
```diff
@@ -111,3 +114,24 @@ func ValidateDeploymentName(n string) error {
...
+
+func (opts *DeploymentOpts) PostRunMessages() error {
+	if !opts.IsCliAuthenticated() {
+		_, err := log.Warningln("To get output for both local and Atlas clusters, run \"atlas login\" command to authenticate your Atlas account.")
+		if err != nil {
+			return err
+		}
+	}
+
+	if err := podman.Installed(); err == podman.ErrPodmanNotFound {
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/options/deployment_opts.go:126]:
```suggestion
	if err := podman.Installed(); errors.Is (err, podman.ErrPodmanNotFound)
```

File: internal/cli/atlas/deployments/options/deployment_opts.go [hunk 6]
```diff
@@ -111,3 +114,24 @@ func ValidateDeploymentName(n string) error {
...
+		_, err = log.Warningln("To get output for both local and Atlas clusters, install Podman.")
+		if err != nil {
+			return err
+		}
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/options/deployment_opts.go:130]:
```suggestion
		if _, err = log.Warningln("To get output for both local and Atlas clusters, install Podman."); err != nil {
			return err
		}
```

File: internal/cli/atlas/deployments/list.go (hunk 4 above)
Comment (blva) [reply internal/cli/atlas/deployments/list.go:199]: for all deployments commands, we won't want to fail if Podman is not ready considering that we'll also try to search for  Atlas deployments

Comment (andreaangiolillo) [reply internal/cli/atlas/deployments/list.go:200]: Thanks for the clarification

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**